File: php/features.php
<?php
declare(strict_types=1);

// PostaliApiRest SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PostaliApiRestFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PostaliApiRestBaseFeature();
            case "test":
                return new PostaliApiRestTestFeature();
            default:
                return new PostaliApiRestBaseFeature();
        }
    }
}

File: py/features.py
# PostaliApiRest SDK feature factory

from feature.base_feature import PostaliApiRestBaseFeature
from feature.test_feature import PostaliApiRestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PostaliApiRestBaseFeature(),
        "test": lambda: PostaliApiRestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
